code/mission/missionparse.cpp

```diff
@@ -3967,7 +3975,8 @@ int parse_wing_create_ships( wing *wingp, int num_to_create, int force, int spec
 				wingp->flags |= WF_WING_GONE;
 
 				// if the current wave is zero, it never existed
-				wingp->flags |= WF_NEVER_EXISTED;
+				if (wingp->current_wave == 0)
+					wingp->flags |= WF_NEVER_EXISTED;
```

Comment:
WF_NEVER_EXISTED does match its function, but it is only checked in one place -- the sexp_query_has_yet_to_arrive helper function -- in a way that appears redundant.

Actually, this situation is rather complicated and carries some interesting consequences.  See #379.